lora/lorawan/region/au915.go

```diff
@@ -7,27 +7,56 @@ const (
 	AU915_DEFAULT_TX_POWER_DBM = 20
 )
 
+type ChannelAU struct {
```

Comment:
This looks like identical code for all ChannelXX types- 
Should we have a base `channel` type and compose these "higher" level types with it?

Reply:
ie:
```go
type channel struct {
	frequency       uint32
	bandwidth       uint8
	spreadingFactor uint8
	codingRate      uint8
	preambleLength  uint16
	txPowerDBm      int8
}
// Implement methods for channel here...

type ChannelAU struct {
     channel // Exposes exported channel methods.
}

Reply:
@soypat that sounds like an excellent refactoring for a subsequent PR.

Reply:
I will submit a PR that does exactly this later on...

Reply:
See https://github.com/tinygo-org/drivers/pull/611